pkg/agent/datamodel/types.go

```diff
@@ -676,6 +676,7 @@ type KubernetesConfig struct {
 	MaximumLoadBalancerRuleCount      int               `json:"maximumLoadBalancerRuleCount,omitempty"`
 	PrivateAzureRegistryServer        string            `json:"privateAzureRegistryServer,omitempty"`
 	NetworkPluginMode                 string            `json:"networkPluginMode,omitempty"`
+	EbpfDataplane                     int               `json:"ebpfDataplane,omitempty"`
```

Comment:
should this be a bool instead?

maybe an enum of you can have other values than `1`?